### PR TITLE
HHS: show blue banner

### PR DESCRIPTION
**Pull request by ganfra**
Allows to handle the first point of #2547
Requires https://github.com/matrix-org/matrix-android-sdk/pull/356

### CODE REVIEW

File: vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+     */
+    init {
+        serverNoticesRooms = loadServerNoticeRooms()
+        serverNoticesRooms.forEach {
+            processPinnedEvents(it)
+        }
+    }
+
+    override fun onRoomTagEvent(roomId: String) {
+        serverNoticesRooms = loadServerNoticeRooms()
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: Yes it's needed, that's a miss

File: vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    object Normal : LimitResourceState()
+    data class Exceeded(val roomId: String, val eventId: String, val matrixError: MatrixError) : LimitResourceState()
+}
+
+/**
+ * This class is responsible for listening to system alert rooms and looking for pinning events.
+ * When an EVENT_TYPE_SERVER_NOTICE_USAGE_LIMIT is pinned, the state goes from Normal to Exceeded.
+ * When the event is unpinned, the state goes back to Normal.
+ */
+class ResourceLimitEventListener(private val dataHandler: MXDataHandler, private val callback: Callback?) : MXEventListener() {
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: Nope, the case is already managed through onRoomTagEvent.

File: vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import org.matrix.androidsdk.data.RoomTag
+import org.matrix.androidsdk.listeners.MXEventListener
+import org.matrix.androidsdk.rest.callback.SimpleApiCallback
+import org.matrix.androidsdk.rest.model.Event
+import org.matrix.androidsdk.rest.model.MatrixError
+import org.matrix.androidsdk.rest.model.ServerNoticeUsageLimitContent
+import org.matrix.androidsdk.util.JsonUtils
+import org.matrix.androidsdk.util.Log
+
+private const val MAX_PINNED_NOTICES_PER_ROOM = 2
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: From Riot Web code!

File: vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+/**
+ * This class represents the possible states the listener can have (Normal or Exceeded)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: It means the limit has not been reached

File: vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt [hunk 5]
```diff
@@ -36,6 +36,11 @@ private const val MAX_PINNED_NOTICES_PER_ROOM = 2
 sealed class LimitResourceState {
     object Normal : LimitResourceState()
     data class Exceeded(val roomId: String, val eventId: String, val matrixError: MatrixError) : LimitResourceState()
+
+    fun softErrorOrNull(): MatrixError? = when (this) {
+        is Exceeded -> matrixError
+        is LimitResourceState.Normal -> null
```
Comment (bmarty) [inline vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: Remove `LimitResourceState.` and put normal first to keep the same order that the declaration

File: vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (manuroe) [reply vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: FTR, riot-web has this [comment](https://github.com/matrix-org/matrix-react-sdk/blob/4b45c17f59d6320493c1c3c0884f7497bd3487ea/src/components/structures/LoggedInView.js#L38-L41) about it.

[the remark this answers is not in the document]
Comment (bmarty) [reply vector/src/main/java/im/vector/features/hhs/ResourceLimitEventListener.kt]: Ok, thanks, I will report this comment in our source

**Pull request merged**